File: 806. Number of Lines To Write String/solution.js
/**
 * @param {number[]} widths
 * @param {string} s
 * @return {number[]}
 */
 var numberOfLines = function(widths, s) {
  let row = 1;
  let currentPixels = 0;

  for (let letter of s){
    let pixels = widths[letter.charCodeAt(0) - 97];
    currentPixels += pixels;
    if (currentPixels > 100){
      row += 1;
      currentPixels = pixels;
    }
  }

  return [row, currentPixels];
};

File: 806. Number of Lines To Write String/solution.py
class Solution(object):
  def numberOfLines(self, widths, s):
    """
    :type widths: List[int]
    :type s: str
    :rtype: List[int]
    """
    row = 1
    currentPixels = 0
    
    for letter in s:
      pixels = widths[ord(letter) - 97]
      currentPixels += pixels
      if currentPixels > 100:
        row += 1
        currentPixels = pixels
      
    return [row, currentPixels]